### ISSUE TITLE
Override functionName of CustomDashresourceDashapigwDashcwDashroleLambdaFunction

### ISSUE THREAD

**Issue by srashish** [serverless/serverless#7744]
I'm facing issues while trying to override function name of CustomDashresourceDashapigwDashcwDashroleLambdaFunction which gets generated when using API Logs in serverless yaml under provider section. 

I've seen that it follows the name convention - `${self:service}-${self:custom.stage}-custom-resource-apigw-cw-role`, I'm trying to override it to use something like this `${self:service}-${self:custom.prefix, self:custom.stage}-custom-resource-apigw-cw-role`

Initial error while deploying

`  An error occurred: ApiGatewayLogGroup - /aws/api-gateway/feature-service-dev already exists.`

Tried overriding this with renaming LogGroupName under resources

serverless.yml

` logs:
    restApi: true
    iamrole:  arn:aws:iam::XXXXXXXXXX:role/serverlessApiGatewayCloudWatchRole`

` 
resources:
  extensions:
    ApiGatewayLogGroup:
      Properties:
        LogGroupName: /aws/apigateway/${self:service}-${self:custom.prefix, self:custom.stage}
`

The generated CF template json which this section is below

`    "CustomDashresourceDashapigwDashcwDashroleLambdaFunction": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": "XXXXXXX",
          "S3Key": "serverless/service-name/dev/xxxxxxxxxxx-2020-05-18T12:46:39.248Z/custom-resources.zip"
        },
        "FunctionName": "feature-service-dev-custom-resource-apigw-cw-role",
        "Handler": "apiGatewayCloudWatchRole/handler.handler",
        "MemorySize": 1024,
        "Runtime": "nodejs12.x",
        "Timeout": 180,
        "Role": {
          "Fn::GetAtt": [
            "IamRoleCustomResourcesLambdaExecution",
            "Arn"
          ]
        }
      },
      "DependsOn": [
        "IamRoleCustomResourcesLambdaExecution"
      ]
    },`

**Comment by medikoo** [comment]
> I'm trying to override it

Why are you trying to override it, what is the use case?

**Comment by srashish** [comment]
@medikoo I'm trying to deploy alias stacks with api logs. If a stack is already using one of the logs then if can't be reused and throws this error --> `An error occurred: ApiGatewayLogGroup - /aws/api-gateway/service-name-dev already exists.`

**Comment by medikoo** [comment]
> @medikoo I'm trying to deploy alias stacks with api logs.

What is _alias stacks_ ?

**Comment by srashish** [comment]
@medikoo So I'm trying to deploy different stacks with same dev settings by changing stage param value but keeping all the settings for dev stage.

**Comment by medikoo** [comment]
> So I'm trying to deploy different stacks

Are they _different_ in sense that they share different service name?

**Comment by srashish** [comment]
@medikoo Yes, I'm tweaking service name by passing extra params while keeping other dependent settings to stage.

**Comment by medikoo** [comment]
> Yes, I'm tweaking service name by passing extra params while keeping other dependent settings to stage.

What's the `service` setting in `serverless.yml` of those services?

**Comment by srashish** [comment]
@medikoo So I have settings based on stage, I've declared two stages as of now, dev and prod. I'm trying to download multiple stacks using dev settings but tweaking their name to use my opt:param rather than default stage param.

Everything goes fine, till I wanted to use logs, since then I'm facing the trouble as my work-around with naming convention it's being applying to it. So in order to make it use my convention, I made some changes in Resources section, but the moment I did that, I stumbled upon this FunctionName issue using this mapping ${self:service}-${self:custom.prefix, self:custom.stage}-custom-resource-apigw-cw-role
